### DISCUSSION TITLE
How to disable user data directory permission changes

### DISCUSSION THREAD

**Discussion by jbking93-git** [felddy/foundryvtt-docker#153] [Q&A] [answered]
Hey felddy.

First off, thanks for a great container, I moved to this after official release and it's been rock solid.

I've noticed the start up script sets permissions on the user data folder at every start.

I have my whole Tabletop resources folder mapped to /Data/resources for easy access to my maps and tokens I own. my Foundry Docker and UnRaid seem to be in a battle for permissions of this folder, as sometimes when I restart my Foundry Docker it takes some time for it to check the permissions and reassign them as it's quite a large folder. This also causes some issues when browsing the folders as a user and I have to manually override them to get access to my folders as a user again.

Am I just doing it wrong? Can I disable setting the user data permissions?

**Comment by felddy** [comment, marked as answer]
Hey @jbking93-git,

I'm glad you've found the container useful!   

You can prevent the container from modifying the permissions of the `/data` volume using the `CONTAINER_PRESERVE_OWNER` environment variable.   Here is the relevant section from the [`README.md`](https://github.com/felddy/foundryvtt-docker#optional):

| Name  | Purpose  |
|-------|---------|
| `CONTAINER_PRESERVE_OWNER` | Normally the ownership of the `/data` directory and its contents are changed to match that of the server at startup.  Setting this to a regular expression will exclude any matching paths and preserve their ownership.   _Note: This is a match on the whole path, not a search._  This is useful if you want mount a volume as read-only inside `/data` (e.g.; a volume that contains assets mounted at `/data/Data/assets`).  |

For your use case set `CONTAINER_PRESERVE_OWNER`  to `/data/Data/resources`

I think this will solve the permission fighting issue for you.  Let me know how it goes.

**Reply by boideltier** [reply]
Hi! I had a smiliar, if not same problem:

I access the `/Data` directory outside of the container via SMB share. As per instructions, I linked the directories via the compose file:
```
volumes:
    - /.../foundryuserdata/:/data
```
Before starting up the container, I made sure that the user who accesses the SMB share has write permissions in the `foundryuserdata` directory. I tried to use `CONTAINER_PRESERVE_OWNER` like this:
```
environment:
    - CONTAINER_PRESERVE_CONFIG=/data/Data
```
Still after loading up the container, it changed the permissions on my `foundryuserdata` directory to read only for the user accessing via SMB. Did I do something wrong?

**Reply by felddy** [reply]
@The-Kangaroo You're using the wrong environment variable.  
You want `CONTAINER_PRESERVE_OWNER`, not `CONTAINER_PRESERVE_CONFIG`.

**Reply by boideltier** [reply]
Ooooh... It's been a long day. Sorry for that. 

Well, it does raise another problem when trying to start my world now. 
`Error: EACCES: permission denied, open '/data/Data/worlds/worldname/data/actors.db'`

I think I'll just try and get by without the share. ^^
